libraries/AP_OSD/AP_OSD_Screen.cpp

```diff
@@ -1749,6 +1769,33 @@ void AP_OSD_Screen::draw_pluscode(uint8_t x, uint8_t y)
 }
 #endif
 
+/*
+  support callsign display from a file called callsign.txt
+ */
+void AP_OSD_Screen::draw_callsign(uint8_t x, uint8_t y)
+{
+#if HAVE_FILESYSTEM_SUPPORT
+    if (!callsign_data.loaded) {
+        callsign_data.loaded = true;
```

Comment:
the way you're using this `loaded` variable is misleading. It's always true, even if loading failed. Needs a new name like `init_has_run_once`

Reply:
fixed